wallet/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -648,10 +682,15 @@ removeHistoryCache cWalId = wsHistoryCache . at cWalId .= Nothing
 removeAccount :: WebTypes.AccountId -> Update ()
 removeAccount accId = wsAccountInfos . at accId .= Nothing
 
+-- | Legacy version of 'removeWAddress2' for backwards compatibility on
+--   the event log.
+removeWAddress :: WebTypes.CWAddressMeta -> Update ()
+removeWAddress = removeWAddress2 . cwamToWam
+
 -- | Remove given address, not removing it completely, but marking it as `removed` instead.
 -- See also 'addRemovedAccount'.
-removeWAddress :: WAddressMeta -> Update ()
-removeWAddress addrMeta@(view wamAccount -> accId) = do
+removeWAddress2 :: WAddressMeta -> Update ()
+removeWAddress2 addrMeta@(view wamAccount -> accId) = do
```

Comment:
Similar idea. We can do this automatically if we define `WAddressMeta` as a safecopy V1 type that migrates from `WebTypes.CWAddressMeta` using the migration code you have below (currently `cwamToWam`).